dashboard/test/helpers/pd/workshop_survey_results_helper_test.rb

```diff
@@ -39,108 +39,31 @@ class Pd::WorkshopSurveyResultsHelperTest < ActionView::TestCase
-    assert_equal 3.0, result_hash[:how_clearly_presented]
-    assert_equal ['Ernie did not put down the ducky', 'Ernie was disorganized'], result_hash[:things_facilitator_could_improve]
-
-    result_hash = summarize_workshop_surveys(workshops: workshops, facilitator_breakdown: false)
-    assert_equal 3.67, result_hash[:how_clearly_presented]
-
-    result_hash = summarize_workshop_surveys(workshops: workshops, facilitator_breakdown: false, include_free_response: false)
-    assert_nil result_hash[:things_facilitator_did_well]
-  end
-
   test 'averaging across multiple surveys' do
-    workshop_1 = create :summer_workshop, num_sessions: 1, num_facilitators: 2, num_completed_surveys: 5
-    workshop_2 = create :summer_workshop, num_sessions: 1, num_facilitators: 3, num_completed_surveys: 10
+    workshop_1 = create :workshop, :teachercon, num_sessions: 1, num_facilitators: 2, num_completed_surveys: 5
+    workshop_2 = create :workshop, :teachercon, num_sessions: 1, num_facilitators: 3, num_completed_surveys: 10
+
+    workshop_1.survey_responses.each do |response|
+      response.update_form_data_hash(
+        {
+          how_happy_after:  'Extremely happy'
+        }
+      )
+      response.save
+    end
 
     workshop_2.survey_responses.each do |response|
       response.update_form_data_hash(
         {
-          how_clearly_presented: {
-            workshop_2.facilitators.first.name => 'Not at all clearly',
-            workshop_2.facilitators.second.name => 'Not at all clearly',
-            workshop_2.facilitators.third.name => 'Not at all clearly'
-          }
+          how_happy_after: 'Not at all happy'
         }
       )
       response.save
     end
 
-    # With 10 people saying "Not at all clearly" to 3 facilitators, and 5 people saying
-    # "Extremely Clearly" to 2 facilitators, we'd expect the answer to be
-    # [(10 * 3 * 1) + (5 * 2 * 5)] / (10 + 30) = 2
+    # With 5 people saying "Extremely happy" (5 out of 5 score) and 10 people saying "Not at all happy" (1 out of 5 score),
+    # we'd expect the answer to be [(10 * 1) + (5 * 5)] / (15 surveys) = 2.33
     result_hash = summarize_workshop_surveys(workshops: [workshop_1, workshop_2], facilitator_breakdown: false)
-    assert_equal 2, result_hash[:how_clearly_presented]
-  end
-
-  test 'get an error if summarizing a mix of workshop surveys' do
-    local_workshop = create(:pd_local_summer_workshop_survey).pd_enrollment.workshop
-    local_workshop.update(course: Pd::Workshop::COURSE_CSP, subject: Pd::Workshop::SUBJECT_CSP_SUMMER_WORKSHOP)
-
-    teachercon = create(:pd_teachercon_survey).pd_enrollment.workshop
-    teachercon.update(course: Pd::Workshop::COURSE_CSP, subject: Pd::Workshop::SUBJECT_CSP_TEACHER_CON)
-    assert_raise RuntimeError do
-      summarize_workshop_surveys(workshops: [local_workshop, teachercon])
-    end
+    assert_equal 2.33, result_hash[:how_happy_after]
   end
```

Comment:
Reason for this test being modified a lot: The test previously used the how_clearly_presented question from the Local Summer workshop survey, which was a facilitator-specific numbered/scaled question (i.e. asking about a specific facilitator of the ones that led the workshop, and the question has worded responses that map to scores of 1-5). Since I had to switch to the TeacherCon survey (the only remaining type that summarize_workshop_surveys works with), I had to pick a different question but the TeacherCon survey has no facilitator-specific numbered/scaled questions, only non-facilitator-specific ones. So, I arbitrarily picked the how_happy_after one which meant the average was now based on total surveys rather than total survey responses about each facilitator.

Reply:
Thanks for the attention to detail here. So I make sure I'm following- all this is under the same 'test averaging across multiple surveys', right? Just having a hard time reading it😂

Reply:
Yes it is! Sorry I should've just linked the first line of it rather than the whole diff